spec/spec_helper.rb

```diff
@@ -81,7 +81,107 @@
     # (e.g. via a command-line flag).
     config.default_formatter = 'doc'
   end
+
+  # Check for leaky test resources.
+  #
+  # Execute this after the test has finished
+  # teardown and mock verifications.
+  #
+  # Changing this to `config.after(:each)` would
+  # put this code inside the test scope, interfering
+  # with the test execution.
+  config.around(:each) do |example|
+    example.run.tap do
+      # TODO: Temporarily disabled for `spec/ddtrace/workers`
+      # was meaningful changes are required to address clean
+      # teardown in those tests.
+      # They currently flood the output, making our test
+      # suite output unreadable.
+      next if example.file_path.start_with?('./spec/ddtrace/workers/async_spec.rb')
```

Comment:
I agree with separating out these changes. Do you think it's worth maybe adding a visible note somewhere (e.g. maybe call `RSpec.warning("FIXME: Async spec tests leak threads")` so it's a bit more visible?

I'm thinking if for some reason we need to divert our efforts we may forget that this is here :)

Reply:
Added explicit warning now.